android/java/org/chromium/chrome/browser/ntp/BraveNtpAdapter.java

```diff
@@ -0,0 +1,610 @@
+        }
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view;
+        if (viewType == TYPE_STATS) {
+            view = LayoutInflater.from(parent.getContext())
+                           .inflate(R.layout.brave_stats_layout, parent, false);
+            return new StatsViewHolder(view);
+
+        } else if (viewType == TYPE_TOP_SITES) {
+            return new TopSitesViewHolder(mMvTilesContainerLayout);
+
+        } else if (viewType == TYPE_NEW_CONTENT) {
+            view = LayoutInflater.from(parent.getContext())
+                           .inflate(R.layout.brave_news_load_new_content, parent, false);
+            return new NewContentViewHolder(view);
+
+        } else if (viewType == TYPE_IMAGE_CREDIT) {
+            view = LayoutInflater.from(parent.getContext())
+                           .inflate(R.layout.ntp_image_credit, parent, false);
+            return new ImageCreditViewHolder(view);
+
+        } else if (viewType == TYPE_NEWS_OPTIN) {
+            view = LayoutInflater.from(parent.getContext())
+                           .inflate(R.layout.optin_layout, parent, false);
+            return new NewsOptinViewHolder(view);
+
+        } else if (viewType == TYPE_NEWS_LOADING) {
+            view = LayoutInflater.from(parent.getContext())
+                           .inflate(R.layout.news_loading, parent, false);
+            return new NewsLoadingViewHolder(view);
+
+        } else {
+            view = LayoutInflater.from(parent.getContext())
+                           .inflate(R.layout.brave_news_row, parent, false);
+            return new NewsViewHolder(view);
+        }
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        int statsCount = getStatsCount();
+        int topSitesCount = getTopSitesCount();
+
+        if (position == 0 && statsCount == 1) {
+            return TYPE_STATS;
+        } else if (topSitesCount == 1 && position == statsCount) {
+            return TYPE_TOP_SITES;
+        } else if (position == statsCount + topSitesCount && mIsNewContent) {
+            return TYPE_NEW_CONTENT;
+        } else if ((position == statsCount + topSitesCount && !mIsNewContent)
+                || (position == statsCount + topSitesCount + 1 && mIsNewContent)) {
+            return TYPE_IMAGE_CREDIT;
+        } else if (position == statsCount + topSitesCount + 1 && mIsDisplayNewsOptin
+                && !mIsNewContent) {
+            return TYPE_NEWS_OPTIN;
+        } else if (position == statsCount + topSitesCount + 1 && shouldDisplayNewsLoading()
+                && !mIsNewContent) {
+            return TYPE_NEWS_LOADING;
+        } else {
+            return TYPE_NEWS;
+        }
+    }
+
+    public int getStatsCount() {
+        return isStatsEnabled() ? 1 : 0;
+    }
+
+    private boolean isStatsEnabled() {
+        return true;
```

Comment:
Is this for future usage here? As it's just never false.

Reply:
yes @samartnik it's for upcoming work for privacy hub. here we are exposing function. i will add a  comment above the function.